src/Chapter2.hs

```diff
@@ -842,7 +879,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n l
+  | n < 0 = []
+  | otherwise = take (length l) (drop n (cycle l))
```

Comment:
That's the correct solution! Though, it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face: